pallets/parachain-staking/src/lib.rs

```diff
@@ -1555,15 +1583,82 @@ pub mod pallet {
             growth_index: GrowthPeriodIndex,
             total_staked: BalanceOf<T>,
             staking_reward_paid_in_era: BalanceOf<T>,
+            total_points: RewardPoint,
+            current_collator_scores: Vec<CollatorScore<T::AccountId>>,
         ) {
             <Growth<T>>::mutate(growth_index, |info| {
                 info.number_of_accumulations = info.number_of_accumulations.saturating_add(1);
                 info.total_stake_accumulated =
                     info.total_stake_accumulated.saturating_add(total_staked);
                 info.total_staker_reward =
                     info.total_staker_reward.saturating_add(staking_reward_paid_in_era);
+                info.total_points = info.total_points.saturating_add(total_points);
+                info.collator_scores =
+                    Self::update_collator_scores(&info.collator_scores, current_collator_scores);
             });
         }
+
+        fn update_collator_scores(
+            existing_collator_scores: &Vec<CollatorScore<T::AccountId>>,
+            current_collator_scores: Vec<CollatorScore<T::AccountId>>,
+        ) -> Vec<CollatorScore<T::AccountId>> {
+            let mut current_scores = existing_collator_scores
+                .into_iter()
+                .map(|current_score| (current_score.collator.clone(), current_score.points.clone()))
+                .collect::<BTreeMap<_, _>>();
+
+            current_collator_scores.into_iter().for_each(|new_score| {
+                current_scores
+                    .entry(new_score.collator)
+                    .and_modify(|points| {
+                        *points = points.saturating_add(new_score.points);
+                    })
+                    .or_insert(new_score.points);
+            });
+
+            return current_scores
+                .into_iter()
+                .map(|(acc, pts)| CollatorScore::new(acc, pts))
+                .collect()
+        }
+
+        pub fn payout_collators(amount: BalanceOf<T>, growth_period: u32) -> DispatchResult {
+            // We do not validate anything here, we trust that the instruction to pay from T1 is ok
+
+            let pay = |collator_address: T::AccountId, amount: BalanceOf<T>| {
```

Comment:
Why do we need to define this function at this level, and not outside? 
I think it makes it harder to understand what is going on, and don't understand its justification.

Reply:
Its a single scope function so there is no need for it to be defined outside. Using a closure allows us to encapsulate the logic in one place.